### ISSUE TITLE
Framework: flavour support (multiple ports from same code with different dependencies)

### ISSUE THREAD

**Issue by fichtner** [opnsense/plugins#2408]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues and I'm convinced that mine is new.
- [x] When the request is meant for an existing plugin, I've added its name to the title.

**Is your feature request related to a problem? Please describe.**

In https://github.com/opnsense/plugins/pull/2116 @fraenki proposed a master/slave concept to allow different Zabbix version plugins in parallel without copying code.

**Describe the solution you'd like**

Instead of master/slave and makefile overrides we would hope to offer a flavoured approach to plugins much like FreeBSD ports offers.

**Describe alternatives you've considered**

Copy plugin and let it bitrot. Ugh. :)

**Additional context**

N/A

**Comment by fichtner** [comment]
Zabbix4-proxy was folded into zabbix-proxy. Zabbix5-proxy we can fold into zabbix-proxy too after a proper release with the new-style build of zabbix4-proxy. In this case zabbix4-proxy gains a few features that previously only zabbix5-proxy had.

Relevant tools commit: https://github.com/opnsense/tools/commit/f1b1f75f909e

Plugins dev workflow is the same unless you want to build the non-default variant of a plugin:

    # cd net-mgmt/zabbix-proxy && make upgrade PLUGIN_VARIANT=zabbix4

CC @fraenki @mimugmail

**Comment by fichtner** [comment]
Stray commits are: fe11c45336dcf8b12c676ec5de5576bc4339201b b9e936f46414d056978f15be17905118bdae2fe7 12b943c41a911e6d701be43085e3033d19f5952a
